src/booleans.ts

```diff
@@ -0,0 +1,26 @@
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/lib/Either";
+import * as t from "io-ts";
+
+export type BooleanFromString = t.Type<boolean, string, unknown>;
+
+export const BooleanFromString: BooleanFromString = new t.Type<
+  boolean,
+  string,
+  unknown
+>(
+  "BooleanFromString",
+  t.boolean.is,
+  (u, c) =>
+    pipe(
+      t.string.validate(u, c),
+      E.chain(s =>
+        s === "true"
+          ? t.success(true)
+          : s === "false"
+          ? t.success(false)
+          : t.failure(u, c)
+      )
+    ),
+  String
+);
```

Comment:
We can avoid the string validation per se as `s === 'true'` implies `s` is a string. Also, by being explicit with the `I` (input) type, we can avoid non-string to even be passed to the decoder

```suggestion
export const BooleanFromString: BooleanFromString = new t.Type<
  boolean,
  string,
  string
>(
  "BooleanFromString",
  t.boolean.is,
  (s, c) =>
    E.chain(s =>
      s === "true"
        ? t.success(true)
        : s === "false"
        ? t.success(false)
        : t.failure(u, c)
    ),
  String
);
```

Reply:
Yeah, we just got gcanti code. We tried to import io-ts-types but that library was breaking build, so we just copied what we needed.  Anyway, your comment seems legit.

Reply:
In this case we nee to remove `E.chain` too.  Also, it's not a `BooleanFromString` anymore

Reply:
Done in https://github.com/pagopa/io-ts-commons/pull/367/commits/2c9ed4840e10021078ef139f8410f67c38e4e2f6